eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -28,61 +29,73 @@ parameters:
-            az group wait --created --resource-group $RgName
-          displayName: Create resource group and environment variable for tests
+          # figure out what's up with the create resource group above
+        - template: ../../../common/TestResources/deploy-test-resources.yml
+          parameters:
+            Location: ${{ coalesce(parameters.Location, cloudConfig.value.Location, 'westus2') }}
+            ServiceDirectory: '${{ parameters.ServiceDirectory }}'
+            SubscriptionConfiguration: ${{ cloudConfig.value.SubscriptionConfiguration }}
+            ArmTemplateParameters: ${{ cloudConfig.value.ArmTemplateParameters }}
+            AdditionalParameters: $ {{ parameters.AdditionalTestArgs }}
 
-      - template: ../steps/build-test.yml
-        parameters:
-          BuildTargetingString: ${{ parameters.BuildTargetingString }}
-          ServiceDirectory: ${{ parameters.ServiceDirectory }}
-          CoverageArg: $(CoverageArg)
-          EnvVars:
-            AZURE_RUN_MODE: 'Live' #Record, Playback
-            ${{ insert }}: ${{ parameters.EnvVars }}
-          BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
-          AfterTestSteps: ${{ parameters.AfterTestSteps }}
-          PythonVersion: $(PythonVersion)
-          OSVmImage: $(OSVmImage)
-          ToxTestEnv: "whl"
-          AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
-          TestMarkArgument: ${{ parameters.TestMarkArgument }}
-          InjectedPackages: ${{ parameters.InjectedPackages }}
-          BuildDocs: ${{parameters.BuildDocs}}
+        - template: ../steps/build-test.yml
+          parameters:
+            BuildTargetingString: ${{ parameters.BuildTargetingString }}
+            ServiceDirectory: ${{ parameters.ServiceDirectory }}
+            CoverageArg: $(CoverageArg)
+            EnvVars:
+              AZURE_RUN_MODE: 'Live' #Record, Playback
+              ${{ insert }}: ${{ parameters.EnvVars }}
+            BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
+            AfterTestSteps: ${{ parameters.AfterTestSteps }}
+            PythonVersion: $(PythonVersion)
+            OSVmImage: $(OSVmImage)
+            ToxTestEnv: "whl"
+            AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
+            TestMarkArgument: ${{ parameters.TestMarkArgument }}
+            InjectedPackages: ${{ parameters.InjectedPackages }}
+            BuildDocs: ${{parameters.BuildDocs}}
 
-      - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
-        - pwsh: |
-            az group delete --resource-group $env:AZURE_RESOURCEGROUP_NAME -y --no-wait
-          displayName: Delete resource group
-          condition: ne(variables['AZURE_RESOURCEGROUP_NAME'], '')
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
```

Comment:
You can replace this with the [remove-test-resources](https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/jobs/archetype-sdk-tests-jobs.yml#L97-L100) template.